### DISCUSSION TITLE
How do I enable a module

### DISCUSSION THREAD

**Discussion by coding4me2** [uzerpllp/uzerp#260] [Q&A] [answered]
Hi,

Just testing this and I disabled the manufacturing module because I don't do manufacturing but then realised afterwards that stock is part of that which in retail is absolutely required therefore I want to enable it again. If I go to setup -> system admin -> modules and click on manufacturing and in there select enable from the left sidebar. If takes me back to the module grid and the manufacturing module continues to show as disabled. Incidentally any module which is disabled will not enable. I've tried them all just to see.

I'm using the DO install.

I'm stumped :) Any help appreciated.

Thanks

**Comment by steveblamey** [comment]
Hi

I don't think disabling modules has any effect. Perhaps you removed manufacturing from your user role.

You can go to setup > admin > roles, click on the 'Administrator' role in the list and then click edit in the side bar.

Tick Manufacturing, under permissions then save.

Log out and back in to reload your user permissions.

**Comment by coding4me2** [comment]
Hi,

It doesn't appear to have an effect until you log out and log in again. When I initially disabled it the manufacturing menu remained there. When I came back later after the log in timed out and logged back in again it was gone and I've not been able to get it back. I've checked the permissions screen and manufacturing isn't an option:

![image](https://user-images.githubusercontent.com/3052210/170532728-2fdec92e-7cc3-4c19-9e7d-e81b49c34c7e.png)

And here's the setup modules permissions in case there was a missing permission which didn't allow me to enable things but all options under modules are selected:

![image](https://user-images.githubusercontent.com/3052210/170533318-f878ec5d-21a3-4a53-b221-d9919d626d6f.png)

And here's the module list showing it disabled:

![image](https://user-images.githubusercontent.com/3052210/170533569-9d8f27e8-8332-48f5-a26d-dc8a39e79a92.png)

Thanks for looking.

**Comment by steveblamey** [comment, marked as answer]
I've tried to replicate the issue. I think the manufacturing permissions, which also produce the menu, were removed when the module was disabled. The easiest option is to delete your droplet and start again.

We control access by creating roles and removing permissions on those roles instead of touching the module states.

The module configuration does need some attention but it's a low priority for us at the moment.

**Comment by coding4me2** [comment]
No problem Steve will stay away from disable and instead user roles and permissions to control access to modules and their features. I'll create a new droplet for this.

**Reply by steveblamey** [reply]
Thanks for your understanding. :smile:
